core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournalConfig.scala

```diff
@@ -42,15 +43,34 @@ class CassandraJournalConfig(system: ActorSystem, config: Config) extends Cassan
             s"Max $maxTagsPerEvent tags per event is supported."
         )
         tag -> tagId
-    }(collection.breakOut)
+    }.foldLeft((HashMap.empty[String, Int], HashMap.empty[String, Int])) {
+      case ((tagsAcc, prefixAcc), (tag, index)) =>
+        if (tag.startsWith("\"") && tag.endsWith("*\"")) {
+          (tagsAcc, prefixAcc + (tag.substring(1, tag.length - 2) -> index))
+        } else {
+          (tagsAcc + (tag -> index), prefixAcc)
+        }
+    }
   }
 
+  def idForTag(tag: String): Int =
+    tags.get(tag).orElse {
+      tagPrefixes
+        .collectFirst {
+          case (prefix, index) if tag.startsWith(prefix) =>
+            index
+        }
+    }.getOrElse(1)
+
   /**
    * Will be 0 if [[#enableEventsByTagQuery]] is disabled,
    * will be 1 if [[#tags]] is empty, otherwise the number of configured
    * distinct tag identifiers.
    */
-  def maxTagId: Int = if (!enableEventsByTagQuery) 0 else if (tags.isEmpty) 1 else tags.values.max
+  def maxTagId: Int =
+    if (!enableEventsByTagQuery) 0
+    else if (tagPrefixes.isEmpty && tags.isEmpty) 1
+    else (tags.values ++ tagPrefixes.values).max
```

Comment:
that `++` may overwrite corresponding value in `tags`, which doesn't have to be the same
```
math.max(tags.values.max, tagPresfixes.values.max)
```

Reply:
That's what I did first, but `math.max(tags.values.max, tagPrefixes.values.max)` will throw `UnsupportedOperationException` if one of them is empty. 
Not sure I understand your comment. What could be overwritten and why? It's all immutable.

Reply:
can't the two maps contain the same key? perhaps that is an impossible configuration

```
Map("foo" -> 2) ++ Map("foo" -> 1)
```

Reply:
We're merging just the values, so it's gonna be `List(2, 1)`, which is fine.

Reply:
ah, missed that, all good then